### DISCUSSION TITLE
Max serverless function size

### DISCUSSION THREAD

**Discussion by EricBatlle** [vercel/community#2028] [Help] [answered]
### Summary

I'm trying to deploy a function that runs puppeteer-core:
![image](https://user-images.githubusercontent.com/15261880/230898941-f11e7de9-1e8b-4eda-bec4-3110c828b5e3.png)

If I run npm publish --dry-run I can check that the function only has a size of 1.2KB:
![image](https://user-images.githubusercontent.com/15261880/230899041-12ff8997-9166-4880-b946-55153888e28a.png)

But when I try to deploy the function to Vercel it throws me this error: 
![image](https://user-images.githubusercontent.com/15261880/230899109-28d65108-b6e1-4ad8-b8f5-58791cfc7e5e.png)

**Comment by amyegan** [comment, marked as answer]
Hi @EricBatlle! [`npm publish`](https://docs.npmjs.com/cli/v9/commands/npm-publish?v=true) is for publishing a package to the registry, so it works a little differently from a deployment build. If you look into your deployment's [build log](https://vercel.com/docs/concepts/deployments/logs) or run `vercel build` locally, that will give you a better idea of the function size when required packages are included.

You can find some suggested solutions for reducing function size in other discussions, such as
- https://github.com/vercel/community/discussions/103
- https://github.com/vercel/community/discussions/124

If you're using this function to generate social media preview images, you can instead use [Dynamic Open Graph Image Generation](https://nextjs.org/blog/next-13-3#dynamic-open-graph-image-generation) in Next.js 13 or Vercel [Open Graph (OG) Image Generation](https://vercel.com/docs/concepts/functions/edge-functions/og-image-generation#) to create these images.
